import-export-cli/cmd/exportAPI.go

```diff
@@ -106,25 +107,53 @@ func executeExportAPICmd(credential credentials.Credential, exportDirectory stri
 // Exported API will be written to a zip file
 func WriteToZip(exportAPIName, exportAPIVersion, zipLocationPath string, resp *resty.Response) {
 	// Write to file
-	//directory := filepath.Join(exportDirectory, cmdcmdExportEnvironment)
-	// create directory if it doesn't exist
+	//	directory := filepath.Join(exportDirectory, cmdcmdExportEnvironment)
+	// 	create directory if it doesn't exist
 	if _, err := os.Stat(zipLocationPath); os.IsNotExist(err) {
 		err = os.Mkdir(zipLocationPath, 0777)
 		if err != nil {
 			utils.HandleErrorAndExit("Error creating zip archive", err)
 		}
 		// permission 777 : Everyone can read, write, and execute
 	}
+
+	// Create a temp directory to save the original zip from the REST API
+	tmpDir, err := ioutil.TempDir("", "apim")
+	if err != nil {
+		_ = os.RemoveAll(tmpDir)
+		utils.HandleErrorAndExit("Error creating a temp folder to keep the original exported zip.", err)
+	}
+
 	zipFilename := exportAPIName + "_" + exportAPIVersion + ".zip" // MyAPI_1.0.0.zip
-	pFile := filepath.Join(zipLocationPath, zipFilename)
-	err := ioutil.WriteFile(pFile, resp.Body(), 0644)
-	// permission 644 : Only the owner can read and write.. Everyone else can only read.
+	tempZipFile := filepath.Join(tmpDir, zipFilename)
+	// Save the zip file in the temp directory.
+	//	Permission 644 : Only the owner can read and write.. Everyone else can only read.
+	err = ioutil.WriteFile(tempZipFile, resp.Body(), 0644)
 	if err != nil {
-		utils.HandleErrorAndExit("Error creating zip archive", err)
+		utils.HandleErrorAndExit("Error creating the original zip archive from the REST API response", err)
 	}
+
+	// Now, we need to extract the zip, copy api_params.yaml file inside and then create the zip again
+	//	First, create a temp directory (tmpClonedLoc) by extracting the original zip file.
+	tmpClonedLoc, err := utils.GetTempCloneFromDirOrZip(tempZipFile)
+	// Create the api_params.yaml file inside the cloned directory.
+	tmpLocationForAPIParamsFile := filepath.Join(tmpClonedLoc, utils.ParamFileAPI)
+	err = impl.ScaffoldAPIParams(tmpLocationForAPIParamsFile)
+	if err != nil {
+		utils.HandleErrorAndExit("Error creating api_params.yaml inside the exported zip archive", err)
+	}
+
+	// Finally, zip the full content.
+	exportedFinalZip := filepath.Join(zipLocationPath, zipFilename)
+	err = utils.Zip(tmpClonedLoc, exportedFinalZip)
+	if err != nil {
+		utils.HandleErrorAndExit("Error creating the final zip archive", err)
+	}
+
```

Comment:
Above unzip, update and zip logic is duplicated in `WriteAPIProductToZip()` function as well. It would be good if we can extract this logic into a separate function and reuse it in both `WriteToZip()` and  `WriteAPIProductToZip()`

Reply:
Added a common function impl.IncludeParamsFileToZip() from https://github.com/wso2/product-apim-tooling/pull/412/commits/93cd1a77638acf9ee748c7b4042442ac30f40f50